os/pm/pm_changestate.c

```diff
@@ -227,6 +233,42 @@ int pm_changestate(enum pm_state_e newstate)
 	 * drivers may refuse the state change.
 	 */
 	if (newstate != PM_RESTORE) {
+
+		/* If new state is same as current state then do nothing*/
+		if (newstate == g_pmglobals.state) {
+			ret = OK;
+			goto EXIT;
+		}
+
+		/* If LCD invokes the state change then change the LCD Power */
+		else if (newstate == PM_NORMAL || newstate == PM_IDLE || (g_pmglobals.state == PM_IDLE && newstate == PM_STANDBY)) {
+			if(newstate == PM_NORMAL) {
+				power = CONFIG_LCD_NORMAL_POWER;
+			}
+			else if(newstate == PM_IDLE) {
+				power = CONFIG_LCD_IDLE_POWER;
+			}
+			else {
+				power = 0;
+			}
+			/* Change LCD Power */
+			fd = open(CONFIG_LCD_DEVPATH, O_RDWR | O_SYNC, 0666);
```

Comment:
For excute change LCD backlight in here.
you can use hpwork queue to avoid excuting it in ISR